gcc/rust/resolve/rust-toplevel-name-resolver-2.0.cc

```diff
@@ -71,6 +71,68 @@ TopLevel::visit (AST::Module &module)
 	      module.get_name ());
 }
 
+void
+TopLevel::visit (AST::ExternCrate &crate)
+{
+  CrateNum num;
+  if (!Analysis::Mappings::get ()->lookup_crate_name (
+	crate.get_referenced_crate (), num))
+    rust_unreachable ();
+
+  auto attribute_macros
+    = Analysis::Mappings::get ()->lookup_attribute_proc_macros (num);
+
+  auto bang_macros = Analysis::Mappings::get ()->lookup_bang_proc_macros (num);
+
+  auto derive_macros
+    = Analysis::Mappings::get ()->lookup_derive_proc_macros (num);
```

Comment:
I think we'll keep a `mappings` member in the `TopLevel` resolver too, so this might end up being simpler soon

Reply:
We probably should use another PR for this change right ?

Reply:
yes, correct